.circleci/config.yml

```diff
@@ -18,6 +18,12 @@ parameters:
   release_build:
     type: boolean
     default: false
+  app_name:
+    type: string
+    default: "WordPress"
+  product_key:
+    type: string
+    default: "wordpress"
```

Comment:
Could be nice to add comments to document those two

Especially I wondered at first if `app_name` had to match something special from the project (must this be the name of the folder in which the app is, aka always `WordPress` for both WP and JP because that's the folder at the repo root? Should it match the name of the product as defined somewhere in some `.properties` or `.gradle` file?…) and realized that in practice it was only used for display in Slack messages and didn't have any impact in the execution of lanes.

Given that the lanes use `app:` as the parameter to distinguish which app to build, I thought `app_name` was supposed to be the value for that parameter at first… until I realized `product_key` was the one for it.

My suggestion is:
 - For this PR, or if not in the next one, add comments to this parameters
 - Optionally, on a future PR (to not block this one, because this is just cleanup and non-blocker), consider renaming those params to avoid confusion with the similar but different names used in the lanes?  (`app_display_name` instead of `app_name`, and `app_key` instead of `product_key`?). This renaming is not critical and definitively for separate PR if ever.

Reply:
Included comments on this! If there's strong feelings about the naming we can address in a separate PR for sure